src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -103,6 +110,91 @@ class IpexCommunicationService extends AgentService {
     });
   }
 
+  async offerAcdc(notificationId: string) {
+    const keriNoti = await this.getNotificationRecordById(notificationId);
+    const msgSaid = keriNoti.a.d as string;
+    const msg = await this.signifyClient.exchanges().get(msgSaid);
+    const schemaSaid = msg.exn.a.s;
+    const creds = await this.signifyClient.credentials().list({
+      filter: {
+        "-s": { $eq: schemaSaid },
+      },
+    });
+    if (!creds || creds.length === 0) {
+      throw new Error(
+        IpexCommunicationService.CREDENTIAL_NOT_FOUND_WITH_SCHEMA
+      );
+    }
+    const pickedCred = creds[0];
+    let holderSignifyName;
+    const holder = await this.identifierStorage.getIdentifierMetadata(
+      msg.exn.a.i
+    );
+    if (holder && holder.signifyName) {
```

Comment:
`signifyName` is a mandatory field now, so this check can be removed.

`const signifyName = (await this.identifierStorage.getIdentifierMetadata(msg.exn.a.i)).signifyName;